### DISCUSSION TITLE
v046-v047 upgrade handler

### DISCUSSION THREAD

**Discussion by RaulBernal** [cosmos/community#46] [Chain Builders (Cosmos SDK)] [answered]
Hello everyone :)

I'm debugging my update_handler for v046->v047 migration (and ibc/go v6.x -> v7.3)

I added extra checks to see what's happening in the background with this: 
https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/simapp/upgrades.go#L30

And this is the result: 
https://github.com/BitCannaGlobal/bcna/blob/BCNA-v3-way-(sdk-v0.47)/app/upgrades.go#L102

### When I run the upgrade:
![image](https://github.com/cosmos/cosmos-sdk/assets/3751926/b2d70484-81c9-4dc5-9db3-b4b80dd32c42)
So I found a subspace that didn't match: `ibc`

If in example, I don't include the `bcna` types  I get (and is normal): 
```No matching subspace found: bcna
KeyTable is not assigned for subspace: bcna
```

But also:
`No matching subspace found: ibc`

What I'm doing wrong? Have I forgotten  the ibc types ?¿?

**Comment by RaulBernal** [comment]
any hint @tac0turtle  & @julienrbrt ?

**Reply by julienrbrt** [reply]
You should ask IBC team. cc @srdtrk

**Reply by RaulBernal** [reply]
Very curious about to reproduce it with simappd (I will try to find the gap)

**Comment by srdtrk** [comment, marked as answer]
Did you add this? @RaulBernal https://github.com/cosmos/ibc-go/blob/c75650a1a037a9fecba5a9005df380f707520ff7/testing/simapp/app.go#L913

**Reply by RaulBernal** [reply]
https://github.com/BitCannaGlobal/bcna/blob/BCNA-v3-way-(sdk-v0.47)/app/app.go#L941

**Reply by RaulBernal** [reply]
Now added to the upgrade handler function ;) 
Thanks for point it

**Reply by RaulBernal** [reply]
Solution:
https://github.com/BitCannaGlobal/bcna/pull/233/files
